### DISCUSSION TITLE
建议新增参数，指定文本分段长度

### DISCUSSION THREAD

**Discussion by Statisticss** [RapidAI/RapidRAG#22] [Feature requests]
多谢大佬分享，很好的开源项目！

问题：使用下来发现，现有文档提取器会将一段文字分割成过多段落，基本上每段只有一句话，上下文之间无法形成关联，导致用户输入query进行匹配时总是匹配不到合适的结果

建议：文档提取器新增参数，用户可以自己根据上传文档的内容特性去指定想要分割文本的平均长度，比如100个字符。当然如果能根据上下文语意去自动分割合适长度的段落就更好了 :)

**Comment by SWHL** [comment]
https://github.com/RapidAI/Knowledge-QA-LLM/blob/73c128042342bac39e01317b44b023a636f397a8/knowledge_qa_llm/config.yaml#L31
这里有参数控制的

**Reply by Statisticss** [reply]
刚看了下这个参数配置的是200，但是好像没有效果？我用tests/test_files/office/word_example.docx做测试，还是会将以下整段文字：

那年冬天，祖母死了，父亲的差使也交卸了，正是祸不单行的日子。我从北京到徐州，打算跟着父亲奔丧回家。到徐州见着父亲，看见满院狼藉的东西，又想起祖母，不禁簌簌地流下眼泪。父亲说：“事已如此，不必难过，好在天无绝人之路！”

拆分为4个段落：

'那年冬天，祖母死了，父亲的差使也交卸了，正是祸不单行的日子。'
'我从北京到徐州，打算跟着父亲奔丧回家。'
'到徐州见着父亲，看见满院狼藉的东西，又想起祖母，不禁簌簌地流下眼泪。'
'父亲说：“事已如此，不必难过，好在天无绝人之路！”'

**Reply by SWHL** [reply]
那估计是有问题了。你这边愿意提个PR来修复一下不

**Reply by Statisticss** [reply]
可以~我先试试看如何解决
